python/lsst/metaserv/metaREST_v0.py

```diff
@@ -24,134 +24,134 @@
+            + "FROM Repo JOIN DbMeta on (repoId=dbMetaId) WHERE dbName = :dbName"
+    message = "No Repos in Database '%s' found." % dbName
+    return _resultOf(text(query), paramMap={"dbName":dbName}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>/tables', methods=['GET'])
 def getDbPerTypeDbNameTables(lsstLevel, dbName):
     '''Lists table names in a given database.'''
-    return runDbQueryM(
-        "SELECT table_name FROM information_schema.tables "
-        "WHERE table_schema= %s ",
-        (dbName,),
-        "No tables found in database '%s'." % dbName)
+    query = "SELECT table_name FROM information_schema.tables WHERE table_schema=:dbName"
+    message = "No tables found in database '%s'." % dbName
+    return _resultsOf(text(query), paramMap={"dbName":dbName}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>/tables/'
                 '<string:tableName>', methods=['GET'])
 def getDbPerTypeDbNameTablesTableName(lsstLevel, dbName, tableName):
     '''Retrieves information about a table from a given database.'''
-    return runDbQuery1(
-        "SELECT DDT_Table.* FROM DDT_Table JOIN DbMeta USING (dbMetaId) "
-        "WHERE dbName=%s AND tableName=%s",
-        (dbName, tableName),
-        "Table '%s.%s'not found." % (dbName, tableName))
+    query = "SELECT DDT_Table.* FROM DDT_Table " \
+            + "JOIN DbMeta USING (dbMetaId) " \
+            + "WHERE dbName=:dbName AND tableName=:tableName"
+    message = "No DDT_Table entries for database '%s'  and table %s." %(dbName, tableName)
+    return _resultOf(text(query), paramMap={"dbName":dbName, "tableName":tableName}, message=message)
 
 @metaREST.route('/db/<string:lsstLevel>/<string:dbName>/' +
                 'tables/<string:tableName>/schema', methods=['GET'])
 def getDbPerTypeDbNameTablesTableNameSchema(lsstLevel, dbName, tableName):
     '''Retrieves schema for a given table.'''
-    return runDbQuery1(
-        "SHOW CREATE TABLE %s.%s" % (dbName, tableName),
-        None,
-        "Table '%s.%s'not found." % (dbName, tableName))
+    # Scalar
+    if SAFE_SCHEMA_PATTERN.match(dbName) and SAFE_TABLE_PATTERN.match(tableName):
+        query = "SHOW CREATE TABLE %s.%s" % (dbName, tableName)
+        message = "Table '%s.%s'not found." % (dbName, tableName)
+        return _resultOf(query, message=message)
+    return _response(_error("ValueError", "Database name or Table name is not safe"))
 
 @metaREST.route('/image', methods=['GET'])
 def getImage():
     return ("meta/.../image not implemented. I am supposed to print list of "
             "supported image types here, something like: raw, template, coadd, "
             "jpeg, calexp, ... etc")
+
+_error = lambda exception, message: {"exception":exception, "message": message}
+_vector = lambda results: {"results": results}
+_scalar = lambda result: {"result": result}
+
+def _resultOf(query, paramMap=None, exception="NotFound",
```

Comment:
`_resultOf` and `_resultsOf` are almost identical, there is a lot of code duplication. Can you merge them into one method?